### PR TITLE
Use title and desc based on message link

### PR DESCRIPTION
**Pull request by teodorus-nathaniel**
# Changes
Make the `[topic]` page allow multiple params, e.g. `/a/b/c/d`. This will allow `/slug/id` to use the same logic as `/slug`.
After the page stops scrolling to the message, it will redirect with shallow true to `/slug`

# Dependencies
- #87

### CODE REVIEW

File: src/pages/[spaceId]/[...topic].tsx [hunk 1]
```diff
@@ -32,27 +33,47 @@ export const getStaticPaths: GetStaticPaths = async () => {
...
+    const topicParams = context.params?.topic as string[]
+    if (topicParams.length <= 0 || topicParams.length > 2) {
+      return undefined
+    }
+
+    const [topic, chatId] = topicParams
+    const roomId = getIdFromSlug(topic)
+    if (!roomId) return undefined
+
+    const invalidChatId = chatId && isNaN(parseInt(chatId))
```
Comment (olehmell) [inline src/pages/[spaceId]/[...topic].tsx:56]:
```suggestion
    const isInvalidChatId = chatId && isNaN(parseInt(chatId))
```

File: src/pages/[spaceId]/[...topic].tsx [hunk 2]
```diff
@@ -32,27 +33,47 @@ export const getStaticPaths: GetStaticPaths = async () => {
...
+    if (invalidChatId) return undefined
```
Comment (olehmell) [inline src/pages/[spaceId]/[...topic].tsx:57]:
```suggestion
    if (isInvalidChatId) return undefined
```

File: src/pages/[spaceId]/[...topic].tsx (hunk 1 above)
Comment (olehmell) [inline src/pages/[spaceId]/[...topic].tsx:56]: I think the code for getting topic you can extract to separate function

**Review by olehmell — CHANGES_REQUESTED** [review]

File: src/pages/[spaceId]/[...topic].tsx (hunk 1 above)
Comment (teodorus-nathaniel) [reply src/pages/[spaceId]/[...topic].tsx]: Updated

**Review by olehmell — APPROVED** [review]

**Pull request merged**